chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -114,196 +152,263 @@ public void updateChebiReferenceMolecules() throws SQLException, Exception
 	 * @throws InvalidAttributeValueException
 	 * @throws Exception
 	 */
-	private void updateMoleculeName(GKInstance molecule, String chebiName, String moleculeName, String prefix) throws InvalidAttributeException, InvalidAttributeValueException, Exception
+	private boolean updateMoleculeName(GKInstance molecule, String chebiName) throws InvalidAttributeException, InvalidAttributeValueException, Exception
 	{
+		String moleculeName = (String) molecule.getAttributeValuesList(ReactomeJavaConstants.name).get(0);
 		if (!chebiName.equals(moleculeName))
 		{
 			molecule.setAttributeValue(ReactomeJavaConstants.name, chebiName);
-			this.nameSB.append(prefix).append(" Old Name: ").append(moleculeName).append(" ; ").append("New Name: ").append(chebiName).append("\n");
-			if (!testMode)
-			{
-				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants.name);
-			}
+			refMolNameChangeLog.info("{}\t{}\t{}\t{}", molecule.getDBID(), molecule.toString() , moleculeName, chebiName);
+			adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants.name);
+			return true;
 		}
+		return false;
 	}
 
 	/**
-	 * Update a molecule's identifier.
+	 * Writes report lines if a molecule's Identifier has changed, according to ChEBI.
 	 * 
 	 * @param molecule
-	 * @param chebiID
-	 * @param moleculeIdentifier
-	 * @param prefix
+	 * @param newChebiID
+	 * @return True if the identifier was updated, false otherwise.
 	 * @throws InvalidAttributeException
 	 * @throws InvalidAttributeValueException
 	 * @throws Exception
 	 */
-	private void updateMoleculeIdentifier(GKInstance molecule, String chebiID, String moleculeIdentifier, String prefix) throws InvalidAttributeException, InvalidAttributeValueException, Exception
+	private void reportIfMoleculeIdentifierChanged(GKInstance molecule, String newChebiID) throws InvalidAttributeException, InvalidAttributeValueException, Exception
 	{
-		if (!chebiID.equals(moleculeIdentifier))
+		String oldMoleculeIdentifier = (String) molecule.getAttributeValue(ReactomeJavaConstants.identifier);
+		if (!newChebiID.equals(oldMoleculeIdentifier))
 		{
-			molecule.setAttributeValue(ReactomeJavaConstants.identifier, chebiID);
-			this.identifierSB.append(prefix).append(" Old Identifier: ").append(moleculeIdentifier).append(" ; ").append("New Identifier: ").append(chebiID).append("\n");
-			if (!testMode)
+			String oldIdentifierReferrersString = "";
+			// Need to get list of DB_IDs of referrers for *old* Identifier and also for *new* Identifier.
+			@SuppressWarnings("unchecked")
+			List<GKInstance> oldIdentifierReferrers = (List<GKInstance>) molecule.getReferers(ReactomeJavaConstants.referenceEntity);
+			if (oldIdentifierReferrers != null && !oldIdentifierReferrers.isEmpty())
 			{
-				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants.identifier);
+				for (GKInstance referrer : oldIdentifierReferrers)
+				{
+					oldIdentifierReferrersString += referrer.getDBID().toString() + "|";
+				}
+			}
+			
```

Comment:
You could do the generation of the oldIdentifierReferrersString like this:
			// Need to get list of DB_IDs of referrers for *old* Identifier and also for *new* Identifier.
			@SuppressWarnings("unchecked")
			String oldIdentifierReferrersString = 
				((List<GKInstance>) molecule.getReferers(ReactomeJavaConstants.referenceEntity))
				.stream()
				.map(referrer -> referrer.getDBID().toString())
				.collect(Collectors.joining("|"));